### PR TITLE
Fix import for cloudflare_list resource

### PR DESCRIPTION
**Pull request by Nmishin**
Fixes https://github.com/cloudflare/terraform-provider-cloudflare/issues/2573

### CODE REVIEW

File: internal/sdkv2provider/resource_cloudflare_list.go [hunk 1]
```diff
@@ -94,10 +94,6 @@ func resourceCloudflareListRead(ctx context.Context, d *schema.ResourceData, met
 	d.Set("description", list.Description)
 	d.Set("kind", list.Kind)
 
-	if !d.HasChange("items") && len(d.Get("item").(*schema.Set).List()) == 0 {
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_list.go:97]: this is to handle cases where the list items are handled externally - https://github.com/cloudflare/terraform-provider-cloudflare/pull/2295#discussion_r1139459258. have you tested that case here?

Comment (Nmishin) [reply internal/sdkv2provider/resource_cloudflare_list.go:97]: Thank you for pointing me to that discussion, didn't see it, and will test it also.

Comment (Nmishin) [reply internal/sdkv2provider/resource_cloudflare_list.go:97]: ok, i understand. Maybe we can to add some description about how that works, and also why I can't delete items if I delete them already from the cloudflare list resources?
Also, maybe we want to add some flag for addition check that if import run, that we want to add items to state?

Comment (Nmishin) [reply internal/sdkv2provider/resource_cloudflare_list.go]: @jacobbednarz, I updated the MR, wdyt? Updating import will allow us to trigger read properly, and don't change any other logic outside of import.

Comment (jacobbednarz) [reply internal/sdkv2provider/resource_cloudflare_list.go]: works for me!

**Comment by jacobbednarz** [comment]
acceptance tests all passing

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareList_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareList_Exists
--- PASS: TestAccCloudflareList_Exists (10.84s)
=== RUN   TestAccCloudflareList_UpdateDescription
--- PASS: TestAccCloudflareList_UpdateDescription (11.16s)
=== RUN   TestAccCloudflareList_Update
--- PASS: TestAccCloudflareList_Update (47.83s)
=== RUN   TestAccCloudflareList_UpdateIgnoreIPOrdering
--- PASS: TestAccCloudflareList_UpdateIgnoreIPOrdering (13.24s)
=== RUN   TestAccCloudflareList_RemoveInlineConfig
--- PASS: TestAccCloudflareList_RemoveInlineConfig (16.86s)
=== RUN   TestAccCloudflareList_Import
--- PASS: TestAccCloudflareList_Import (13.04s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	113.884s
```

**Pull request merged**